start.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+"""
+Programmatically launch modules
+"""
+import argparse
+import select
+import subprocess
+
+from modules import data_types, load_config
+
+
+# Add to the launch list any modules that can be run
+# TAG: module + startup commands
+launch_list = {
+    "rs_depth": ["modules.realsense", "-o", "rs_depth"],
+    "rs_pose": ["modules.realsense", "-o", "rs_pose"],
+    "rs_color": ["modules.realsense", "-o", "rs_color"],
+}
+# ---DO NOT MODIFY BELOW THIS LINE---
+
+
+def parse_cli() -> argparse.Namespace:
+    """
+    Run argparse on the input arguments to find modules and config file
+    """
+    parser = argparse.ArgumentParser(description="Startup module(s)")
+    parser.add_argument(
+        "-m",
+        "-M",
+        "--module",
+        action="append",
+        nargs="+",
+        default=None,
+        help="[Module Name] [Debug] [Config File] [Process Name]",
+    )
+    parser.add_argument(
+        "-c",
+        "-C",
+        "--config-file",
+        type=str,
+        default=None,
+        help="Startup list config file",
+    )
+    parser.add_argument(
+        "--except-capture",
+        action="store_true",
+        required=False,
+        default=False,
+        help="Enable subprocess exception capture",
+    )
+    parser.add_argument(
+        "--wait-on-process",
+        action="store_true",
+        required=False,
+        default=False,
+        help="Wait for processes to close before exiting",
+    )
+    return parser.parse_known_args()[0]
+
+
+def check_if_tag_known(tag: str) -> bool:
+    """
+    Check if the tag is in the launch dictionary
+    """
+    if tag in launch_list.keys():
+        return True
+    else:
+        print("TAG:{} not known".format(tag))
+        return False
+
+
+def parse_startup_yaml(config_file: str, current_startup_list=None) -> list:
+    """
+    Convert yaml startup file into a list of StartupItem
+    """
+    if current_startup_list is None:
+        current_startup_list = []
+
+    conf = load_config.from_file(config_file, use_cli_input=False)
+
+    for _ in conf.startup_list:
+        if not check_if_tag_known(_.module):
+            continue
+
+        current_startup_list.append(
+            load_config.conf_to_named_tuple(data_types.StartupItem, _)
+        )
+
+    return current_startup_list
+
+
+def parse_startup_cli(modules: list, current_startup_list=None) -> list:
```

Comment:
I feel like there is a better way to do this. I need to dig out an old shell script but I think it would be better to actually look at each of these with a flag, rather than location on the cli (unless I have completely misunderstood this).
Makes it a bit more robust.

Reply:
I can't another way of robustly doing this, positional arguments seems to be the only way with argparse

Reply:
https://stackoverflow.com/questions/10448200/how-to-parse-multiple-nested-sub-commands-using-python-argparse